### PR TITLE
Create iam-limits.adoc

### PR DESCRIPTION
**Pull request by sarathecoubian**

### CODE REVIEW

**Review by gkop — DISMISSED** [review]

File: modules/ROOT/pages/iam-limits.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+= Limits in Access Management
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Access Management imposes the following limits on various features in an organization:
+
+* Roles per client: 3000
+* Clients per root organization: 10000
```
Comment (gkop) [inline modules/ROOT/pages/iam-limits.adoc]: This seems maybe incorrect to me.

```
ms_authentication> select org_id, count(*) as num_clients from clients group by org_id having count(*) >= 10000 order by count(*) desc;
+--------------------------------------+---------------+
| org_id                               | num_clients   |
|--------------------------------------+---------------|
| ee85d185-ed55-419a-8ca0-fda90872691d | 14182         |
| 99838324-71ae-4681-94f3-5425c6ed5ffd | 10001         |
| 315666df-0540-4afc-a4c3-158c5275ab91 | 10000         |
| b2473331-fff9-45aa-b3f4-f8dfba5d20df | 10000         |
+--------------------------------------+---------------+

ms_authentication> select sub_organization_ids from organizations where id = '7f34e134-ef86-4a88-a34a-368f44ff5920';
-[ RECORD 1 ]-------------------------
sub_organization_ids | {3e646885-33af-42ab-987f-b602c33841f7,129d1685-60c9-4eab-b506-5042c06012c9,b486fec7-3340-4b7e-b5fa-27fee86def4c,d0da16a1-974f-4fb0-baa4-9149c2e5a586}

ms_authentication> select created_at from clients where org_id in ('3e646885-33af-42ab-987f-b602c33841f7'::uuid,'129d1685-60c9-4eab-b506-5042
 c06012c9'::uuid,'b486fec7-3340-4b7e-b5fa-27fee86def4c'::uuid,'d0da16a1-974f-4fb0-baa4-9149c2e5a586'::uuid) order by created_at desc limit 10
 ;
+----------------------------+
| created_at                 |
|----------------------------|
| 2021-11-10 16:12:07.211681 |
| 2021-11-10 06:39:40.796051 |
| 2021-11-09 23:54:26.233054 |
| 2021-11-08 16:10:27.194359 |
| 2021-11-05 11:11:52.938565 |
| 2021-11-05 10:39:09.826806 |
| 2021-11-05 09:41:00.744449 |
| 2021-11-05 09:31:04.193474 |
| 2021-11-01 11:23:52.974624 |
| 2021-11-01 09:08:59.927887 |
+----------------------------+
```

So you can see the customer is able to still create clients on the BGs, even though the root org is at its limit.

File: modules/ROOT/pages/iam-limits.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* Environments per organization: 150
+* Sub organizations per root organization: 100
+* Roles per organization: 500
+* External access relations per organization: 20
+* Teams per root organization: 1000
+* Users per organization: 10000
+* Roles per role group: 400
+* Roles per user: 5000
+* Role groups per user: 700
+* Connected apps per organization: 200
```
Comment (gkop) [inline modules/ROOT/pages/iam-limits.adoc]: Where is this defined in the code/config? I don't see 200 as a value here:
```
authentication-server git:(integration/develop) git grep CONNECTED api/config/external_properties.env | grep COLLEC
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_CONNECTEDAPPLICATIONS_SCOPES_ENABLED=[boolean:true]
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_CONNECTEDAPPLICATIONS_SCOPES_SLATIERS_LIMIT=[number:1000]
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_ORGANIZATIONS_CONNECTEDAPPLICATIONS_INTERNAL_ENABLED=[boolean:true]
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_ORGANIZATIONS_CONNECTEDAPPLICATIONS_INTERNAL_SLATIERS_LIMIT=[number:2000]
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_ORGANIZATIONS_CONNECTEDAPPLICATIONS_PUBLIC_ENABLED=[boolean:true]
api/config/external_properties.env:AUTH_SERVER_COLLECTION_LIMITING_POLICIES_ORGANIZATIONS_CONNECTEDAPPLICATIONS_PUBLIC_SLATIERS_LIMIT=[number:20]

➜  authentication-server git:(integration/develop) git grep 200\] api/config/external_properties.env
api/config/external_properties.env:AUTH_SERVER_MIGRATIONS_20190126166006_USER_ROLES_POPULATE_JSONB_TARGETTIME=[number:1200]
api/config/external_properties.env:AUTH_SERVER_MIGRATIONS_MASTERORGTEAMSMIGRATION_BATCHSIZE=[number:200]
api/config/external_properties.env:AUTH_SERVER_RATELIMITING_SLATIERS_CLIENTS_FIND_CALLERROUTE_INTERNAL_MAXREQUESTSPERWINDOW=[number:200]
api/config/external_properties.env:AUTH_SERVER_RATELIMITING_SLATIERS_IPROUTECHEAP_EXTERNAL_MAXREQUESTSPERWINDOW=[number:200]
api/config/external_properties.env:AUTH_SERVER_RATELIMITING_SLATIERS_ORGANIZATIONS_GETHIERARCHY_CALLERROUTE_INTERNAL_MAXREQUESTSPERWINDOW=[number:200]
```

I checked kilonova and there are no overrides (yay).

cf. https://mulesoft.slack.com/archives/C025U9CR49W/p1636652826015000?thread_ts=1636630156.008800&cid=C025U9CR49W

Comment (gkop) [reply modules/ROOT/pages/iam-limits.adoc]: https://mulesoft.slack.com/archives/C02FTAMUK9Q/p1637630854088800?thread_ts=1637630002.088700&cid=C02FTAMUK9Q

Comment (sarathecoubian) [reply modules/ROOT/pages/iam-limits.adoc]: David sent me the following, which is how I got all the numbers I used:

[
  { name: 'Roles per Client', limit: 3000 },
  { name: 'Clients per Organization', limit: 10000 },
  { name: 'Client Management Providers per Organization', limit: 10 },
  { name: 'Environments per Organization', limit: 150 },
  { name: 'SubOrgs per Master Org', limit: 100 },
  { name: 'Roles per Organization', limit: 500 },
  { name: 'External Access Relations per Organization', limit: 20 },
  { name: 'Teams per Master Organization', limit: 1000 },
  { name: 'Users per Organization', limit: 10000 },
  { name: 'Roles per Role Group', limit: 400 },
  { name: 'Roles per User', limit: 5000 },
  { name: 'Role Groups per User', limit: 700 },
  { name: 'Connected Applications Per Organization', limit: 200 },
  { name: 'Scope Assignments per Connected Application', limit: 1000 },
  { name: 'Authorized Applications per User', limit: 30 },
  { name: 'Role assignments per Team', limit: 1000 },
  { name: 'External group mappings per team', limit: 100 }
]

File: modules/ROOT/pages/iam-limits.adoc (hunk 1 above)
Comment (dwang8) [reply modules/ROOT/pages/iam-limits.adoc]: can this be changed to `Clients per root organization or sub organization`

File: modules/ROOT/pages/iam-limits.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Teams, roles, and user permissions:
+
+* Teams per root organization: 1000
+* Permission roles per team: 1000 -> Permissions per team?
+* Rolegroups/deprecated roles per user: 700 -> Roles per user?
+* Permission roles per api manager client or runtime client: 3000 -> ??
+* Deprecated roles per root or sub organization: 500 -> ??
+* Permission roles per deprecated role: 400 ->??
+* Permission roles per user: 5000 ->??
```
Comment (sarathecoubian) [inline modules/ROOT/pages/iam-limits.adoc]:
```suggestion
* Permission roles per user: 5000 ->?? individual permissions?
```

File: modules/ROOT/pages/iam-limits.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+* Permission roles per external organization: 1000 -> ??
+
+Teams, roles, and user permissions:
+
+* Teams per root organization: 1000
+* Permission roles per team: 1000 -> Permissions per team?
+* Rolegroups/deprecated roles per user: 700 -> Roles per user?
+* Permission roles per api manager client or runtime client: 3000 -> ??
+* Deprecated roles per root or sub organization: 500 -> ??
+* Permission roles per deprecated role: 400 ->??
```
Comment (sarathecoubian) [inline modules/ROOT/pages/iam-limits.adoc]:
```suggestion
* Permission roles per deprecated role: 400 -> Permissions each role can have associated with it? 
```

File: modules/ROOT/pages/iam-limits.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+* External access relations per root or sub organization: 20
+* Permission roles per external organization: 1000 -> ??
+
+Teams, roles, and user permissions:
+
+* Teams per root organization: 1000
+* Permission roles per team: 1000 -> Permissions per team?
+* Rolegroups/deprecated roles per user: 700 -> Roles per user?
+* Permission roles per api manager client or runtime client: 3000 -> ??
+* Deprecated roles per root or sub organization: 500 -> ??
```
Comment (sarathecoubian) [inline modules/ROOT/pages/iam-limits.adoc]:
```suggestion
* Deprecated roles per root or sub organization: 500 -> ?? Roles each org can have configured?
```

File: modules/ROOT/pages/iam-limits.adoc [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+* Identity management providers per root or sub organization: 25
+
+Client providers:
+
+* Client management providers per root or sub organization: 10
+
+Connected apps:
+
+* Authorized applications per user: 30 -> Connected apps per user? or.
+* External connected applications per root or sub organization: 20 -> Connected Apps shared using the External Access feature?
```
Comment (sarathecoubian) [inline modules/ROOT/pages/iam-limits.adoc]:
```suggestion
* External connected applications per root or sub organization: 20 -> Connected Apps accessible by external organizations.
```

**Pull request merged**